src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -232,27 +246,28 @@ public function placeAction(Request $request)
      * @return Response
      */
     #[AdminSecurity("is_granted('create', request.get('_legacy_controller'))")]
-    public function createAction(Request $request)
-    {
-        /** @var ShopContext $shopContextChecker */
-        $shopContextChecker = $this->get('prestashop.adapter.shop.context');
-
-        if (!$shopContextChecker->isSingleShopContext()) {
+    public function createAction(
+        Request $request,
+        LanguageByIdChoiceProvider $choiceProvider,
+        #[Autowire(service: 'prestashop.core.form.choice_provider.currency_by_id')] FormChoiceProviderInterface $currencies,
+    ) {
+        $isSingleShopContext = $this->getShopContext()->getShopConstraint()->getShopId() !== null;
+        if (!$isSingleShopContext) {
             $this->addFlash('error', $this->trans(
                 'You have to select a shop before creating new orders.',
+                [],
                 'Admin.Orderscustomers.Notification'
             ));
 
             return $this->redirectToRoute('admin_orders_index');
         }
 
         $summaryForm = $this->createForm(CartSummaryType::class);
-        $languages = $this->get('prestashop.core.form.choice_provider.language_by_id')->getChoices(
+        $languages = $choiceProvider->getChoices(
             [
-                'shop_id' => $shopContextChecker->getContextShopID(),
+                'shop_id' => $this->getShopContext()->getShopConstraint()->getShopId(),
             ]
         );
-        $currencies = $this->get(CurrencyByIdChoiceProvider::class)->getChoices();
```

Comment:
Careful you changed the value of `$currencies` from an array result to the service `FormChoiceProviderInterface`
It should be `$currencies = $currencyChoiceProvider->getChoices()`